flang/test/Lower/OpenMP/parallel-reduction-mixed.f90

```diff
@@ -4,6 +4,10 @@
 ! RUN: %flang_fc1 -emit-llvm -fopenmp -o - %s 2>&1 \
 ! RUN: | FileCheck %s
 
+! UNSUPPORTED: target-x86
+! UNSUPPORTED: target=sparc-{{.*}}
+! UNSUPPORTED: target=sparcel-{{.*}}
```

Comment:
Why is this test unsupported on those platforms?

Reply:
I disabled them because they just so happen to invoke `malloc` and thus no longer match exactly on these 32-bit platforms. This is because `malloc` now gets called with an `i32`. I figured it was okay since 32-bit wasn't officially supported or being tested by the public builders.

Reply:
Thanks for your quick response. If possible I would like to keep the test coverage. For these tests the test doesn't really care what type is used with malloc, so long as the right number is used. So you could do something like
```
!CHECK: call ptr @malloc(i{{[0-9]+}} 80)
```

Reply:
I just fixed the OpenMP-related tests. `character-1.f90` remains disabled as it really does dynamically allocate memory, and the resulting additional `trunc` instructions cannot easily be matched.

As a result, this PR now includes some drive-by fixes in `XReboxOpConversion` as well as inside `OpenMPIRBuilder::createReductions` because, ironically, `parallel-reduction-mixed.f90` seems as if it never worked correctly on 32-bit (calling `__kmpc_reduce` with the wrong type).